### DISCUSSION TITLE
error pterodactyl curseforge not starting

### DISCUSSION THREAD

**Discussion by arturxcvv** [pterodactyl/panel#3811] [Support]
**this happens when I start the server containing the: forge-1.12.2-14.23.5.2860-installer
in the server folder**

[Pterodactyl Daemon]: Checking server disk space usage, this could take a few seconds...
[Pterodactyl Daemon]: Updating process configuration files...
[Pterodactyl Daemon]: Ensuring file permissions are set correctly, this could take a few seconds...
container@pterodactyl~ Server marked as starting...
[Pterodactyl Daemon]: Pulling Docker container image, this could take a few minutes to complete...
Pulling from pterodactyl/yolks 
Digest: sha256:6d84884bc3667c426abbf62d866fb42812d90de2e868978adc958057fad66ebf 
Status: Image is up to date for ghcr.io/pterodactyl/yolks:java_16 
[Pterodactyl Daemon]: Finished pulling Docker container image
container@pterodactyl~ java -version
openjdk version "16.0.2" 2021-07-20
OpenJDK Runtime Environment (build 16.0.2+7-67)
OpenJDK 64-Bit Server VM (build 16.0.2+7-67, mixed mode, sharing)
container@pterodactyl~ java -Xms128M -Xmx8192M -jar server.jar
Error: Unable to access jarfile server.jar
container@pterodactyl~ Server marked as offline...

[Pterodactyl Daemon]: ---------- Detected server process in a crashed state! ----------
[Pterodactyl Daemon]: Exit code: 1
[Pterodactyl Daemon]: Out of memory: false
[Pterodactyl Daemon]: Checking server disk space usage, this could take a few seconds...
[Pterodactyl Daemon]: Updating process configuration files...
[Pterodactyl Daemon]: Ensuring file permissions are set correctly, this could take a few seconds...
container@pterodactyl~ Server marked as starting...
[Pterodactyl Daemon]: Pulling Docker container image, this could take a few minutes to complete...
Pulling from pterodactyl/yolks 
Digest: sha256:6d84884bc3667c426abbf62d866fb42812d90de2e868978adc958057fad66ebf 
Status: Image is up to date for ghcr.io/pterodactyl/yolks:java_16 
[Pterodactyl Daemon]: Finished pulling Docker container image
container@pterodactyl~ java -version
openjdk version "16.0.2" 2021-07-20
OpenJDK Runtime Environment (build 16.0.2+7-67)
OpenJDK 64-Bit Server VM (build 16.0.2+7-67, mixed mode, sharing)
container@pterodactyl~ java -Xms128M -Xmx8192M -jar server.jar
Error: Unable to access jarfile server.jar
container@pterodactyl~ Server marked as offline...

[Pterodactyl Daemon]: ---------- Detected server process in a crashed state! ----------
[Pterodactyl Daemon]: Exit code: 1
[Pterodactyl Daemon]: Out of memory: false
[Pterodactyl Daemon]: Aborting automatic restart, last crash occurred less than 60 seconds ago.

**when renaming the jar file in: server.jar, I get the following error**

[Pterodactyl Daemon]: Checking server disk space usage, this could take a few seconds...
[Pterodactyl Daemon]: Updating process configuration files...
[Pterodactyl Daemon]: Ensuring file permissions are set correctly, this could take a few seconds...
container@pterodactyl~ Server marked as starting...
[Pterodactyl Daemon]: Pulling Docker container image, this could take a few minutes to complete...
Pulling from pterodactyl/yolks 
Digest: sha256:6d84884bc3667c426abbf62d866fb42812d90de2e868978adc958057fad66ebf 
Status: Image is up to date for ghcr.io/pterodactyl/yolks:java_16 
[Pterodactyl Daemon]: Finished pulling Docker container image
container@pterodactyl~ java -version
openjdk version "16.0.2" 2021-07-20
OpenJDK Runtime Environment (build 16.0.2+7-67)
OpenJDK 64-Bit Server VM (build 16.0.2+7-67, mixed mode, sharing)
container@pterodactyl~ java -Xms128M -Xmx8192M -jar server.jar
JVM info: Oracle Corporation - 16.0.2 - 16.0.2+7-67
java.net.preferIPv4Stack=true
Found java version 16.0.2
Exception in thread "main" 
java.awt.HeadlessException: 
No X11 DISPLAY variable was set,
but this program performed an operation which requires it.
        at java.desktop/java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:165)
        at java.desktop/java.awt.Window.<init>(Window.java:549)
        at java.desktop/java.awt.Frame.<init>(Frame.java:426)
        at java.desktop/java.awt.Frame.<init>(Frame.java:391)
        at java.desktop/javax.swing.SwingUtilities$SharedOwnerFrame.<init>(SwingUtilities.java:1918)
        at java.desktop/javax.swing.SwingUtilities.getSharedOwnerFrame(SwingUtilities.java:1994)
        at java.desktop/javax.swing.JOptionPane.getRootFrame(JOptionPane.java:1689)
        at java.desktop/javax.swing.JOptionPane.showOptionDialog(JOptionPane.java:868)
        at java.desktop/javax.swing.JOptionPane.showMessageDialog(JOptionPane.java:670)
        at java.desktop/javax.swing.JOptionPane.showMessageDialog(JOptionPane.java:641)
        at net.minecraftforge.installer.SimpleInstaller.launchGui(SimpleInstaller.java:177)
        at net.minecraftforge.installer.SimpleInstaller.main(SimpleInstaller.java:147)
container@pterodactyl~ Server marked as offline...

[Pterodactyl Daemon]: ---------- Detected server process in a crashed state! ----------
[Pterodactyl Daemon]: Exit code: 1
[Pterodactyl Daemon]: Out of memory: false
[Pterodactyl Daemon]: Checking server disk space usage, this could take a few seconds...
[Pterodactyl Daemon]: Updating process configuration files...
[Pterodactyl Daemon]: Ensuring file permissions are set correctly, this could take a few seconds...
container@pterodactyl~ Server marked as starting...
[Pterodactyl Daemon]: Pulling Docker container image, this could take a few minutes to complete...
Pulling from pterodactyl/yolks 
Digest: sha256:6d84884bc3667c426abbf62d866fb42812d90de2e868978adc958057fad66ebf 
Status: Image is up to date for ghcr.io/pterodactyl/yolks:java_16 
[Pterodactyl Daemon]: Finished pulling Docker container image
container@pterodactyl~ java -version
openjdk version "16.0.2" 2021-07-20
OpenJDK Runtime Environment (build 16.0.2+7-67)
OpenJDK 64-Bit Server VM (build 16.0.2+7-67, mixed mode, sharing)
container@pterodactyl~ java -Xms128M -Xmx8192M -jar server.jar
JVM info: Oracle Corporation - 16.0.2 - 16.0.2+7-67
java.net.preferIPv4Stack=true
Found java version 16.0.2
Exception in thread "main" 
java.awt.HeadlessException: 
No X11 DISPLAY variable was set,
but this program performed an operation which requires it.
        at java.desktop/java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:165)
        at java.desktop/java.awt.Window.<init>(Window.java:549)
        at java.desktop/java.awt.Frame.<init>(Frame.java:426)
        at java.desktop/java.awt.Frame.<init>(Frame.java:391)
        at java.desktop/javax.swing.SwingUtilities$SharedOwnerFrame.<init>(SwingUtilities.java:1918)
        at java.desktop/javax.swing.SwingUtilities.getSharedOwnerFrame(SwingUtilities.java:1994)
        at java.desktop/javax.swing.JOptionPane.getRootFrame(JOptionPane.java:1689)
        at java.desktop/javax.swing.JOptionPane.showOptionDialog(JOptionPane.java:868)
        at java.desktop/javax.swing.JOptionPane.showMessageDialog(JOptionPane.java:670)
        at java.desktop/javax.swing.JOptionPane.showMessageDialog(JOptionPane.java:641)
        at net.minecraftforge.installer.SimpleInstaller.launchGui(SimpleInstaller.java:177)
        at net.minecraftforge.installer.SimpleInstaller.main(SimpleInstaller.java:147)
container@pterodactyl~ Server marked as offline...

[Pterodactyl Daemon]: ---------- Detected server process in a crashed state! ----------
[Pterodactyl Daemon]: Exit code: 1
[Pterodactyl Daemon]: Out of memory: false
[Pterodactyl Daemon]: Aborting automatic restart, last crash occurred less than 60 seconds ago.

sorry for the bad english of google translator, and my total ignorance in servers, i just want to turn on a curseforge server on pterodactyl.
![Screenshot_2](https://user-images.githubusercontent.com/74259279/146142871-006219c5-aa2c-4941-8cd9-f67b1e2e3f1d.png)
![Screenshot_3](https://user-images.githubusercontent.com/74259279/146142875-b0e36927-d6fe-4167-8ea6-b153d6ec0ff2.png)

**Comment by trenutoo** [comment]
You can't run an installer like that. It's not a bundled server jar, and instead requires params passed to it or a GUI hence the X11 display error.

The curseforge egg is not able to handle all modpacks, since many of them have different methods of installation. You should install those manually.
